### DISCUSSION TITLE
Conversions, contributions, and customization

### DISCUSSION THREAD

**Discussion by oscarlevin** [MRU-COMP1701-dev/fopp-pretext#19] [General]
Hi all.  I stumbled on this project from the Runestone library page.  I'm teaching an intro to CS through python course this fall and was planning to build up a fork of FOPP that moves functions earlier among other things.  I was also going to convert FOPP to PreTeXt along the way.

Now that I'm seeing your project, I'm thinking it makes more sense to at least start out with what you have built and modify from there.  Maybe.  Is there anything I should watch out for at this point?  I'd be happy to contribute back what I modify in case it is something you would like to fold in, of course.

Since I know they would also like a PreTeXt version of the original FOPP, I want to do that too.  But it looks like you have done a lot already, so if you have any suggestion of how to proceed, that would be wonderful.

So to summarize, here is what I'm thinking of doing, but I'd love to hear your thoughts
1. Fork this book, modify the `document-id` and publish a hidden version just for my students on Runestone.
2. Keep a branch that is basically the same as your version, but any fixes I find, I can contribute back.
3. On the main branch (that pushes to Runestone), add back sections converted from the original FOPP that I want (like dictionaries and turtles).
4. Use your conversion of some sections and my conversions of others to contribute back to FOPP to complete their conversion to PreTeXt.

No idea how best to acknowledge the conversions in the original FOPP, but that seems like something we could figure out later.
